api/src/main/java/io/strimzi/api/kafka/model/status/Status.java

```diff
@@ -45,13 +45,24 @@ public void setConditions(List<Condition> conditions) {
         this.conditions = conditions;
     }
 
-    public void addCondition(Condition condition) {
+    private List<Condition> prepareConditionsUpdate() {
         List<Condition> oldConditions = getConditions();
         List<Condition> newConditions = oldConditions != null ? new ArrayList<>(oldConditions) : new ArrayList<>();
```

Comment:
If we're going to the trouble to size the new list why not pass in the number of things which will be added?

Reply:
As the constructor is using an existing list, rather than a size, this looked a bit fiddly to change so I've left it as-is for now.